### ISSUE TITLE
Replace syslinux theme?

### ISSUE THREAD

**Issue by davidmhewitt** [elementary/os#248]
Opening this as a place to gather thoughts and feedback on a potential change.

On an elementary (or Ubuntu) .iso, there are two bootloaders that can present a boot menu before loading the live session to allow checking the integrity of the disk, or setting kernel parameters in case of weird hardware. 

Syslinux is used for legacy (BIOS) systems and Grub is used for UEFI systems. On the current stable .iso, this looks like the following:

Syslinux timeout (waits 5 seconds here before booting into live session):
![image](https://user-images.githubusercontent.com/3372394/67402728-d64a1180-f5a8-11e9-91e9-beec2b6e3204.png)

Syslinux menu, presented when you press a key in the above 5 second timeout (ignore artifacts, that's my VM doing weird things):
![image](https://user-images.githubusercontent.com/3372394/67402892-2032f780-f5a9-11e9-99c6-20e0b3d8ee2d.png)

The language, keymap and accessibility options are potentially interesting as they allow you to set these things at an early stage in theory. However, I think a lot of these options are extremely untested. For example, there are options in some of these menus for OEM installs, drive update discs, expert mode (whatever that is). This is also completely unmodifiable as it stands because these extra menus are the result of a compiled gfxboot blob that we're currently borrowing from Ubuntu. So removing the OEM install option for example isn't possible without figuring out how to modify and recompile this weird code in a language I've never seen before:
https://git.launchpad.net/ubuntu/+source/gfxboot-theme-ubuntu/tree/common.inc
  
Also, if you're using anything other than a 4:3 display (which I'm sure most people are), that elementary logo is pretty stretched looking.

Here's the current grub bootloader from the stable iso:
![image](https://user-images.githubusercontent.com/3372394/67404167-da772e80-f5aa-11e9-93e5-ec00dd66ab47.png)

We can see that this grub loader doesn't have any of the additional functionality offered by the extra menus in the syslinux loader and it also looks completely different. So we offer pretty untested (but potentially nice) options to people with legacy BIOS systems and not many options to people with UEFI systems.

What I propose doing is making this the syslinux timout menu:
![image](https://user-images.githubusercontent.com/3372394/67394531-443c0c00-f59c-11e9-8ef0-df8d5560dbdb.png)

And this the syslinux menu (accessed by pressing a button during the timeout):
![image](https://user-images.githubusercontent.com/3372394/67394648-751c4100-f59c-11e9-9baf-1b73ecc980a3.png)

It's not as pretty, sure. But it's simple to understand in terms of config files, so we won't need to fork and maintain an Ubuntu syslinux theme (as it's only like 15 lines of config). It doesn't offer options we don't want to support or test and doesn't have a stretched elementary logo.

For comparison this is the grub theme I've built that's shipping in the current daily .iso that gets presented to UEFI users:
![image](https://user-images.githubusercontent.com/3372394/67404963-00510300-f5ac-11e9-8ec0-48f383208214.png)

The logo doesn't get stretched as it's a fixed size 100x100 pixel image rathre than a background that gets stretched to the screen (this doesn't seem to be possible with syslinux). I also still need to add the additional option for checking the disk for defects and one for booting the hard drive (if I can figure out how), along with possibly another option for entering the UEFI config.

Obviously we can tweak the copy for either bootloader or switch back to the grub theme used in the 5.0 .iso if we wish. Integrating the old syslinux theme into the new .iso comes with some difficulties, but wouldn't be impossible if that's the route we wanted to take.

But we can see that this work brings the two bootloaders closer together in terms of functionality (and look to an extent) based on what's possible with each.

Thoughts/comments?

**Comment by cassidyjames** [comment]
I like your proposed changes. I think my main concerns in general are:

1. Maintainability; there are a bunch of moving pieces and we should try to simplify as much as sanely possible.
2. Ability to pass kernel params; this is required for certain hardware to even boot.
3. l10n support. If we have to punt this from the bootloader in favor of handling it in the installer, I could probably be convinced. That doesn't work for the Live session with Ubiquity, but should work with the installer-only session.

**Comment by kgrubb** [comment]
I also like the proposed changes. 

For maintainability, I'd say that setting the syslinux and grub themes in this repo is cleaner and easier to find. It's also already a step forward in regards to grub theming, since previously, theming grub was pretty hidden and hard to set up.

From what I can tell, passing kernel params is still possible, as long as we give a short timeout, so I think that one's safe too. 

I'll defer the i10n support to other people, since I don't have much of an opinion on it.

**Comment by davidmhewitt** [comment]
I've spent the last couple of days trying to get the old Ubuntu syslinux bootloader (with language selection) working with the new method of building the .iso and have been unsuccessful.

I think I've narrowed it down to the fact that the Debian live-build scripts build the .iso using Rock Ridge extensions which allow nicer handling of longer filenames and permissions on the .iso and it seems the `gfxboot` stuff that does the nice menus in syslinux doesn't support this.

I think the conclusion there being that we'd have to switch back to the old Ubuntu `live-build` way of doing things and figuring out why Cody's scripts aren't building a properly working .iso if we want that syslinux menu.
